### DISCUSSION TITLE
CheckBoxList set data

### DISCUSSION THREAD

**Discussion by NazarAli** [filamentphp/filament#9079] [Help]
i have checkboxlist that relationsip field
i want hide it depend on value of another field after hide set data to empty
i use $set but not work with hidden field 

![Screenshot 2023-10-13 173853](https://github.com/filamentphp/filament/assets/7056487/842f7b58-a4b3-4a7e-871f-948aaacd9857)

**Comment by VincvanAs** [comment]
I think you don't have to use Set anymore.

Remove the AfterStateUpdated part, keep the ->reactive() and change your checkboxlist something like:

`->hidden(fn (Get $get) => $get('type_key') !== AppConstant::scheduleWeeklyKey)`

**Reply by NazarAli** [reply]
the problem the hidden form field not updated value after the submit 
i try this code and work 


```
protected function handleRecordUpdate(Model $record, array $data): Model
 {
        $record->update($data);

        if($data['type_key'] == AppConstant::ScheduleWeeklyKey )
        {

            $record->monthDays()->sync([]);
        }else if($data['type_key']== AppConstant::ScheduleMonthlyKey)
        {
            $record->weekDays()->sync([]);
        }
        return $record;
}
```

**Reply by VincvanAs** [reply]
I think replacing reactive with ->live(onBlur: true)

May solve your issue on your original code
